### ISSUE TITLE
Fixing Smushed Chapter Labels in Line Graph

### ISSUE THREAD

**Issue by kes213** [ebeshero/DHClass-Hub#394]
So, I'm getting each number for the chapter to output onto the SVG document, but when I open the document locally to view it, all of the numbers are squished together.

Here's what I have on my XSLT document:
```
<text x="{$Pos * $Xinterval}" y="50" fill="blue" text-anchor="middle"><xsl:apply-templates select="//@which"/></text>
```
I've defined each of my variables, and I've also tried including an extra `$Xinterval div 2` or simply `$Xinterval ` in my attribute value for x, but that had very little to almost no effect on my graph. 

This is what my output graph currently looks like:
<img width="395" alt="capture" src="https://user-images.githubusercontent.com/31480908/32519890-2b9c0964-c3dc-11e7-81c8-0eb30ef2cf6f.PNG">

Does anyone have a suggestion for how I can fix this?

**Comment by pab124** [comment]
This is essentially the same problem that I'm having except that I have `<xsl:apply-templates select="alice/chapter/@which"/>` within the `<text>` element. I figured that it had something to do with how I was trying to grab the @which attributes using the xPath, but in the xPath window, it gives them as individual values. Just curious how to use xPath to retrieve them and display them within the template as unique, multiple entries.

**Comment by ebeshero** [comment]
@kes213 @pab124 Aha! The problem you both are having, I think, has to do with how you're stepping from the context of your `<xsl:for-each select="CONTEXT">`. (So yes, I agree with your diagnosis of the problem, @pab124). Remember, @kes213, that when you use the leading `//` at the start of an expression, that means "go back up to the root node of the document." It does not mean "go down the descendant axis from my context node." I think what's happening is that you're going back to the root node to spit out all the chapters at once, over and over again, 12 times (once for each chapter).

If the `@which` is sitting on the `<chapter>` element, and you've set each `<chapter>` as the context of xsl:for-each, you just need to invoke `<xsl:apply-templates select="@which"/>` inside your for-each loop.

**Comment by kes213** [comment]
I am still struggling. 

When I made the changes discussed in this issue and the other issue that I opened, a lot of things happened, and not all of them were good. My chapters evened out, but I lost the space above my graph, and no matter what numbers I use in transform="translate ()", it doesn't create space at the top. Instead, it cuts out or overstretches the graph. 

Also, my circles appear if I do not multiply the count() of the times Alice speaks per chapter by the Yinterval, but if I do multiply it, some dots disappear entirely, as they do in the photo below. 

`<circle cx="{$Pos * $Xinterval - $Xinterval div 2}" cy="{count(descendant::q[@sp='alice']) * $Yinterval}" r="4" fill="blue"/>`

<img width="435" alt="screenshot" src="https://user-images.githubusercontent.com/31480908/32531676-3f5055dc-c413-11e7-80f0-5a95a29005e6.PNG">

I don't know where to go from here.

**Comment by ebeshero** [comment]
@kes213 I'd like to take a closer look at your code: can you share it somehow? Perhaps push to the Troubleshooting directory, and I'll take a quick look. I'm not sure I can tell from the image what's wrong--but my guess is, one of the following:
* perhaps something's not well-formed in your SVG--it seems like the output code might be off somehow if it's not adjusting for any changes to the dimensions that you set?
* OR, that in correcting some of the XPath, you need to change some of the parameters you set for your viewport and viewBox attributes on the `<svg>` element. (I can't tell if that's the problem, but I'm betting that's something's off. I don't understand why your circles are starting to plot so far to the right, when your hashmarks are perfectly positioned.

**Comment by kes213** [comment]
@ebeshero I pushed the XSLT file to my personal repo.

**Comment by ebeshero** [comment]
@kes213 Wait a minute: I think I understand something of the problem: You must have recalculated a variable determining the length of your y axis. Am I right about that? Look at my sample output: http://dh.newtfire.org/alice_svg_output.svg 

Your plot is now correctly positioning the dots for Chapters 10 and 11, when Alice doesn't speak at all: We can see those are correct. But I think the others are so high as to be off the screen. Let's take a look at the scaling of your graph and scale back your $Yinterval, maybe? I'm taking a look at your code now....

**Comment by ebeshero** [comment]
@kes213 I left some line comments on the code you just committed. Let me know if that helps solve some issues!

**Comment by kes213** [comment]
I'm unable to see your comments . . . 

I did a git pull for my repo, but it says that it's already up to date, and when I open the file locally again, it doesn't have any new comments on it. 

I think what I will do is turn in what I have. I've been working on this fairly consistently since yesterday afternoon, and I think it's time to give my eyes a break from it. I did adjust the Yinterval and my dots reappeared, and but I don't have the energy to attempt to fix the lines right now. 

I'll add my title and labels and call it a night, I think. 

Thank you for all of your help today.
